apis/externalsecrets/v1alpha1/pushsecret_types.go

```diff
@@ -97,6 +98,10 @@ type PushSecretMatch struct {
 type PushSecretData struct {
 	// Match a given Secret Key to be pushed to the provider.
 	Match PushSecretMatch `json:"match"`
+	// Metadata is metadata attached to the secret.
+	// The structure of metadata is provider specific, please look it up in the provider documentation.
+	// +optional
+	Metadata *apiextensionsv1.JSON `json:"metadata,omitempty"`
```

Comment:
We cannot use the `interface{}` directly, yet we can use `apiextensionsv1.JSON` for the purpose.

https://github.com/kubernetes-sigs/controller-tools/pull/126